src/wallet/wallet.cpp

```diff
@@ -2532,7 +2532,7 @@ CAmount CWallet::GetWatchOnlyBalance() const
         for (std::map<uint256, CWalletTx>::const_iterator it = mapWallet.begin(); it != mapWallet.end(); ++it)
         {
             const CWalletTx* pcoin = &(*it).second;
-            if (pcoin->IsTrusted())
+            if (!pcoin->IsTrusted() && pcoin->GetDepthInMainChain() == 0 && !pcoin->IsLockedByInstantSend() && pcoin->InMempool())
```

Comment:
"Fixed" the wrong one here, should not touch this and fix `GetUnconfirmedWatchOnlyBalance()` instead